SofaKernel/modules/SofaCore/src/sofa/core/DataTracker.cpp

```diff
@@ -63,15 +64,15 @@ void DataTracker::clean()
 
 
 ////////////////////
-void DataTrackerDDGNode::addInputs(std::initializer_list<sofa::core::objectmodel::BaseData*> datas)
+void DataTrackerDDGNode::addInputs(std::initializer_list<sofa::core::objectmodel::DDGNode*> datas)
 {
-    for(sofa::core::objectmodel::BaseData* d : datas)
+    for(sofa::core::objectmodel::DDGNode* d : datas)
         addInput(d);
 }
 
-void DataTrackerDDGNode::addOutputs(std::initializer_list<sofa::core::objectmodel::BaseData*> datas)
+void DataTrackerDDGNode::addOutputs(std::initializer_list<sofa::core::objectmodel::DDGNode*> datas)
 {
-    for(sofa::core::objectmodel::BaseData* d : datas)
+    for(sofa::core::objectmodel::DDGNode* d : datas)
```

Comment:
I'm not sure of this part....A data tracker is supposed to track data...otherwise name that a DDGNodeTracker :)

Reply:
Well it's actually a DataTrackerDDGNode... :D but yes I understand your point.
Having DataTrackerDDGNode register DDGNodes instead of BaseData will let us track nodephysics::Links (which are built upon the DDGNode API). But since the DDG Links are not part of this pull request I will remove this change from this PR

Reply:
Taken into consideration in last commit (also applied on #1406 which introduces the code in question)